### ISSUE TITLE
Deprecation warning favors unknown resource

### ISSUE THREAD

**Issue by wernerfred** [hashicorp/terraform-provider-azurerm#17859]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.5

### AzureRM Provider Version

3.15.1

### Affected Resource(s)/Data Source(s)

azurerm_security_center_server_vulnerability_assessment

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_virtual_network" "example" {
  name                = "example-vnet"
  resource_group_name = azurerm_resource_group.example.name
  address_space       = ["192.168.1.0/24"]
  location            = azurerm_resource_group.example.location
}

resource "azurerm_subnet" "example" {
  name                 = "example-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["192.168.1.0/24"]
}

resource "azurerm_network_interface" "example" {
  name                = "example-nic"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  ip_configuration {
    name                          = "vm-example"
    subnet_id                     = azurerm_subnet.example.id
    private_ip_address_allocation = "Dynamic"
  }
}

resource "azurerm_linux_virtual_machine" "example" {
  name                = "example-vm"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  size                            = "Standard_B1s"
  admin_username                  = "testadmin"
  admin_password                  = "Password1234!"
  disable_password_authentication = false

  source_image_reference {
    publisher = "OpenLogic"
    offer     = "CentOS"
    sku       = "7.5"
    version   = "latest"
  }

  os_disk {
    caching              = "ReadWrite"
    storage_account_type = "Standard_LRS"
  }

  network_interface_ids = [azurerm_network_interface.example.id]
}

resource "azurerm_security_center_server_vulnerability_assessment" "example" {
  virtual_machine_id = azurerm_linux_virtual_machine.example.id
}
```

### Debug Output/Panic Output

```shell
╷
│ Warning: Deprecated Resource
│ 
│   with module.defaults.azurerm_security_center_server_vulnerability_assessment.this,
│   on ../../main.tf line 380, in resource "azurerm_security_center_server_vulnerability_assessment" "this":
│  380: resource "azurerm_security_center_server_vulnerability_assessment" "this" {
│ 
│ This resource has been deprecated in favour of the
│ 'azurerm_security_center_vm_server_vulnerability_assessment' resource and
│ will be removed in v4.0 of the AzureRM Provider
╵
```

### Expected Behaviour

The deprecation warning should reference an existing resource. For example instead of `azurerm_security_center_server_vulnerability_assessment` use `azurerm_security_center_server_vulnerability_assessment_virtual_machine`

### Actual Behaviour

The deprecation warning shows that one should use `azurerm_security_center_vm_server_vulnerability_assessment` which isn't available in the registry/provider

### References

#15747

**Comment by liuwuliuyun** [comment]
Hi @wernerfred , could you try azurerm_security_center_server_vulnerability_assessment_virtual_machine to see if it fits your request? Here is the link https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/security_center_server_vulnerability_assessment_virtual_machine

Maybe we dont need to add a new resource but to modify the deprecation warning.

**Comment by liuwuliuyun** [comment]
Since we haven't supported Hybrid Virtual Machines yet [link](https://github.com/hashicorp/terraform-provider-azurerm/issues/15990) I think modify doc and warning line in code to say 'This resource has been deprecated in favour of the 'azurerm_security_center_server_vulnerability_assessment_virtual_machine' resource'

**Comment by liuwuliuyun** [comment]
PR raised to fix this: https://github.com/hashicorp/terraform-provider-azurerm/pull/17874

**Comment by wernerfred** [comment]
> Hi @wernerfred , could you try azurerm_security_center_server_vulnerability_assessment_virtual_machine to see if it fits your request? Here is the link https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/security_center_server_vulnerability_assessment_virtual_machine
> 
> Maybe we dont need to add a new resource but to modify the deprecation warning.

The new resource `azurerm_security_center_server_vulnerability_assessment_virtual_machine` behaves same way as `azurerm_security_center_server_vulnerability_assessment`. Both resources looks exactly same with only 1 difference that [hybrid_machine_id](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/azurerm_security_center_server_vulnerability_assessment#hybrid_machine_id) was removed in new resource. (Which is expected) 
So your linked PR that changes the description is the best solution!

**Comment by rcskosir** [comment]
@wernerfred Thanks for taking the time to submit this issue. It looks like this has been resolved as of #17874. As such, I am going to mark this issue as closed.
